File: src/interfaces/discord/commands/general/about-command.ts
import { AboutCommandUseCase } from "@/application/usecases/general/about-command-usecase";
import { TranslationService } from "@/infrastructure/i18n/translation-service";
import { DiscordGuildCommandCategory } from "@/interfaces/discord/types/i-command";
import type { ChatInputCommandInteraction } from "discord.js";
import { inject, injectable } from "inversify";
import { BaseCommand } from "../base-command";

@injectable()
export class AboutCommand extends BaseCommand {
  constructor(
    @inject(AboutCommandUseCase) private readonly aboutCommandUseCase: AboutCommandUseCase,
    @inject(TranslationService) translate: TranslationService
  ) {
    super({
      name: translate.t("CMD_ABOUT_NAME"),
      description: translate.t("CMD_ABOUT_DESCRIPTION"),
      category: DiscordGuildCommandCategory.GENERAL
    });
  }

  public async execute(interaction: ChatInputCommandInteraction): Promise<void> {
    await this.aboutCommandUseCase.execute(interaction);
  }
}

File: src/main/java/dev/quasemago/aura/interfaces/discord/command/general/AboutCommand.java
package dev.quasemago.aura.interfaces.discord.command.general;

import dev.quasemago.aura.application.usecase.general.AboutCommandUseCase;
import dev.quasemago.aura.infrastructure.i18n.TranslationService;
import dev.quasemago.aura.interfaces.discord.command.AbstractSlashCommand;
import dev.quasemago.aura.interfaces.discord.command.CommandCategory;
import discord4j.core.event.domain.interaction.ChatInputInteractionEvent;
import discord4j.core.object.entity.User;
import discord4j.discordjson.json.ApplicationCommandRequest;
import org.springframework.stereotype.Component;
import reactor.core.publisher.Mono;

@Component
public class AboutCommand extends AbstractSlashCommand {
    private final AboutCommandUseCase useCase;
    private final TranslationService translate;

    public AboutCommand(AboutCommandUseCase useCase, TranslationService translate) {
        this.useCase = useCase;
        this.translate = translate;
    }

    @Override
    public Mono<Void> execute(ChatInputInteractionEvent event, User author) {
        return useCase.execute(event, author);
    }

    @Override
    public String getName() {
        return translate.t("CMD_ABOUT_NAME");
    }

    @Override
    public String getDescription() {
        return translate.t("CMD_ABOUT_DESCRIPTION");
    }

    @Override
    public ApplicationCommandRequest getCommand() {
        return commandBuilder().build();
    }

    @Override
    public CommandCategory getCategory() {
        return CommandCategory.GENERAL;
    }
}
